src/NHibernate/Multi/IQueryBatch.cs

```diff
@@ -1,24 +1,61 @@
-namespace NHibernate
+using System;
+using System.Collections.Generic;
+
+namespace NHibernate.Multi
 {
 	/// <summary>
 	/// Universal query batcher
 	/// </summary>
 	public partial interface IQueryBatch
 	{
 		/// <summary>
-		/// Executes batch
+		/// Executes the batch.
 		/// </summary>
 		void Execute();
-		
+
 		/// <summary>
-		/// Adds query to batch.
+		/// Adds a query to the batch.
 		/// </summary>
-		/// <param name="query">Query</param>
+		/// <param name="query">The query.</param>
+		/// <exception cref="InvalidOperationException">Thrown if the batch has already been executed.</exception>
```

Comment:
And is not in "auto-reset" mode. The batch used for futures is in auto-reset mode.